p2p/conn/connection_test.go

```diff
@@ -769,3 +770,273 @@ func stopAll(t *testing.T, stoppers ...stopper) func() {
 		}
 	}
 }
+
+// generateMessages sends a sequence of messages to the specified multiplex connection `mc`.
+// Each message has the given size and is sent at the specified rate
+// `messagingRate`. This process continues for the duration `totalDuration` or
+// until `totalNum` messages are sent. If `totalNum` is negative,
+// messaging persists for the entire `totalDuration`.
+func generateMessages(mc *MConnection,
+	messagingRate time.Duration,
+	totalDuration time.Duration, totalNum int, msgSize int, chID byte) {
+	// all messages have an identical content
+	msg := bytes.Repeat([]byte{'x'}, msgSize)
+
+	// message generation interval ticker
+	ticker := time.NewTicker(messagingRate)
+	defer ticker.Stop()
+
+	// timer for the total duration
+	timer := time.NewTimer(totalDuration)
+	defer timer.Stop()
+
+	sentNum := 0
+	// generating messages
+	for {
+		select {
+		case <-ticker.C:
+			// generate message
+			if mc.Send(chID, msg) {
+				sentNum++
+				if totalNum > 0 && sentNum >= totalNum {
+					log.TestingLogger().Info("Completed the message generation as the" +
+						" total number of messages is reached")
+					return
+				}
+			}
+		case <-timer.C:
+			// time's up
+			log.TestingLogger().Info("Completed the message generation as the total " +
+				"duration is reached")
+			return
+		}
+	}
+}
+
+func BenchmarkMConnection(b *testing.B) {
+	chID := byte(0x01)
+
+	// Testcases 1-3 evaluate the effect of send queue capacity on message transmission delay.
+
+	// Testcases 3-5 assess the full utilization of maximum bandwidth by
```

Comment:
3 seems like it falls in the previous category so this one starts at 4? 

```suggestion
	// Testcases 4-5 assess the full utilization of maximum bandwidth by
```

Reply:
Oh got it. Sorry I think my suggestions should be ignored.